plugins/modules/nd_pcv.py

```diff
@@ -0,0 +1,211 @@
+    if name is None:
+        nd.existing = ndi.query_pcvs(insights_group)
+    elif site_name is not None:
+        nd.existing = ndi.query_pcv(insights_group, site_name, name)
+
+    if state == 'absent':
+        nd.previous = nd.existing
+        job_id = nd.existing.get('jobId')
+        if nd.existing and job_id:
+            if module.check_mode:
+                nd.existing = {}
+            else:
+                rm_path = '{0}/{1}/prechangeAnalysis/jobs'.format(path, insights_group)
+                rm_payload = [job_id]
+                rm_resp = nd.request(rm_path, method='POST', data=rm_payload, prefix=ndi.prefix)
+                if rm_resp["success"] == True:
+                    nd.existing = {}
+                else:
+                    nd.fail_json(msg="Pre-change validation {0} is not able to be deleted".format(name))
+
+    elif state == 'present':
+        nd.previous = nd.existing
+        # if nd.existing:
+        #     nd.exit_json()
+        base_epoch_data = ndi.get_epochs(insights_group, site_name)
+
+        data = {
+            "allowUnsupportedObjectModification": "true",
+            "analysisSubmissionTime": round(time.time() * 1000),
+            "baseEpochId": base_epoch_data["epochId"],
+            "baseEpochCollectionTimestamp": base_epoch_data["collectionTimeMsecs"],
+            "fabricUuid": base_epoch_data["fabricId"],
+            "description": description,
+            "name": name,
+            "assuranceEntityName": site_name,
+        }
+        if file:
+            if not os.path.exists(file):
+                nd.fail_json(msg="File not found : {0}".format(file))
+            # check whether file content is a valid json
+            if ndi.is_json(open(file, "rb").read()) is False:
+                extract_data = ndi.load(open(file))
+            else:
+                extract_data = json.loads(open(file, "rb").read())
+            if isinstance(extract_data, list):
+                ndi.cmap = {}
+                tree = ndi.construct_tree(extract_data)
+                ndi.create_structured_data(tree, file)
+            create_pcv_path = '{0}/{1}/fabric/{2}/prechangeAnalysis/fileChanges'.format(path, insights_group, site_name)
+            file_resp = nd.request(create_pcv_path, method='POST', file=os.path.abspath(file), data=data, prefix=ndi.prefix)
+            if file_resp.get("success") == True:
+                nd.existing = file_resp.get("value")["data"]
+        elif manual:
+            data["imdata"] = json.loads(manual)
+            create_pcv_path = '{0}/{1}/fabric/{2}/prechangeAnalysis/manualChanges?action=RUN'.format(path, insights_group, site_name)
+            manual_resp = nd.request(create_pcv_path, method='POST', data=data, prefix=ndi.prefix)
+            if manual_resp.get("success") == True:
+                nd.existing = manual_resp.get("value")["data"]
+        else:
+            nd.fail_json(msg="either file or manual is required to create pcv job")
```

Comment:
I think you can use required_one_of for this purpose. See https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec-dependencies

Reply:
only when state == present, we will need either file or manual